pkg/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -3210,8 +3210,103 @@ func vmTests() {
 						Expect(moVM.Snapshot.CurrentSnapshot).ToNot(BeNil())
 
 						// Find the snapshot name in the tree to verify it matches
-						currentSnapName := vmlifecycle.FindSnapshotNameInTree(moVM.Snapshot.RootSnapshotList, moVM.Snapshot.CurrentSnapshot.Value)
-						Expect(currentSnapName).To(Equal(vmSnapshot.Name))
+						currentSnap := vmlifecycle.FindSnapshotInTree(moVM.Snapshot.RootSnapshotList, moVM.Snapshot.CurrentSnapshot.Value)
+						Expect(currentSnap).ToNot(BeNil())
+						Expect(currentSnap.Name).To(Equal(vmSnapshot.Name))
+					})
+				})
+
+				// Simulate an Imported Snapshot scenario by
+				//	- creating the VC VM and VM while ensuring the backup is not taken.
+				//	- creating a snapshot on VC AND THEN only creating the VMSnapshot CR so that the ExtraConfig is
+				// 	  not stamped by the controller.
+				// 	- change some bits in the VM CR and take a second snapshot. This snapshot can be taken through a
+				//	  VMSnapshot. This is needed to make sure that we are not reverting to a snapshot that the VM is
+				//    running off at the same time.
+				//	- Now, revert the VM to the first snapshot. It is expected that the spec fields would now be approximated.
+				Context("when reverting to imported snapshot", func() {
+					var secondSnapshot *vmopv1.VirtualMachineSnapshot
+
+					BeforeEach(func() {
+						pkgcfg.SetContext(parentCtx, func(config *pkgcfg.Config) {
+							config.Features.VMImportNewNet = true
+						})
+					})
+
+					It("should successfully revert to desired snapshot and approximate the VM Spec", func() {
+						if vm.Labels == nil {
+							vm.Labels = make(map[string]string)
+						}
+
+						vm.Labels[kubeutil.CAPVClusterRoleLabelKey] = ""
```

Comment:
What's the goal of this label? To skip reconcileSnapshot?

Reply:
No, to skip backup.